cmake/README.md

```diff
@@ -1,23 +1,101 @@
 
-# CMake
+# CMake Guide
 
 CMake is an open-source, cross-platform family of tools designed to build, test and package software. CMake is used to control the software compilation process using simple platform and compiler independent configuration files, and generate native makefiles and workspaces that can be used in the compiler environment of your choice.
 
-### Links
+## Requirement
 
-* Official website: [cmake.org](https://cmake.org/)
+1. Open your terminal, input `cmake --version`, if the cmake version is lower than 3.1, please upgrade
 
-* Documentation: [cmake.org/documentation](https://cmake.org/documentation/)
+2. You should use out-of-source build, that is said you need to create a different directory from `cocos2d-x` to execute cmake command.
+
+## Step by Step
+
+### Build in Linux
 
-* CMake FAQ: [Wiki/CMake_FAQ](https://cmake.org/Wiki/CMake_FAQ)
+```sh
+cd cocos2d-x
```

Comment:
why we need for each platform Linux/Win32/Mac different instruction?
It should be everywhere the same:
$ cd cocos2d-x
$ mkdir build
$ cd build
$ cmake ..
$ cmake --build .

Reply:
Right, we can do this in Linux/Win32/Mac native build, not include iOS and Android, different instruction tell users other informations, for example how to generate Xcode project.

Specially, cmake build only support MSVC, didn't support MinGw for win32 platform, so I think adding visible option will avoid conflicts.